File: imageviewer/src/main/java/com/king/image/imageviewer/ImageViewerActivity.java
package com.king.image.imageviewer;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.ViewCompat;
import androidx.viewpager2.widget.ViewPager2;

/**
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
public class ImageViewerActivity extends AppCompatActivity {

    static final String SHARED_ELEMENT = "shared_element";

    private TextView tvIndicator;

    ImageViewerAdapter mAdapter;

    ViewerSpec mViewerSpec;

    private int mSize;

    private boolean isShowIndicator;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState, @Nullable PersistableBundle persistentState) {
        super.onCreate(savedInstanceState, persistentState);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewerSpec = ViewerSpec.INSTANCE;
        setRequestedOrientation(mViewerSpec.orientation);
        setTheme(mViewerSpec.theme);
        setContentView(R.layout.image_viewer_activity);
        init();
    }

    private void init(){
        tvIndicator = findViewById(R.id.tvIndicator);
        ViewPager2 viewPager = findViewById(R.id.viewPager);

        viewPager.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                super.onPageScrolled(position, positionOffset, positionOffsetPixels);
            }

            @Override
            public void onPageSelected(int position) {
                super.onPageSelected(position);
                if(isShowIndicator){
                    updateIndicator(position,mSize);
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {
                super.onPageScrollStateChanged(state);
            }
        });

        ViewCompat.setTransitionName(viewPager,SHARED_ELEMENT);

        mAdapter = new ImageViewerAdapter(mViewerSpec.listData);
        viewPager.setAdapter(mAdapter);

        mAdapter.setOnItemClickListener(new ImageViewerAdapter.OnItemClickListener() {
            @Override
            public void onClick(View v, int position) {
                onBackPressed();
            }
        });


        int position = mViewerSpec.position;
        mSize = mAdapter.getItemCount();
        if(position>=0){
            viewPager.setCurrentItem(position,false);
            updateIndicator(position,mSize);
        }

        isShowIndicator = mViewerSpec.isShowIndicator && mSize > 0;
        if(isShowIndicator){
            tvIndicator.setVisibility(View.VISIBLE);
        }
    }

    private void updateIndicator(int position,int size){
        tvIndicator.setText(String.format("%s/%s",Math.min(position + 1,size),size));
    }

    @Override
    protected void onDestroy() {
        if(mViewerSpec != null ){
            mViewerSpec.imageLoader = null;
        }
        super.onDestroy();
    }
}

File: imageviewer/src/main/java/com/king/image/imageviewer/ImageViewerActivity.kt
package com.king.image.imageviewer

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.viewpager2.widget.ViewPager2
import androidx.viewpager2.widget.ViewPager2.OnPageChangeCallback

/**
 * 浏览图片时的界面
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 * <p>
 * <a href="https://github.com/jenly1314">Follow me</a>
 */
open class ImageViewerActivity : AppCompatActivity() {

    private var tvIndicator: TextView? = null

    private lateinit var mAdapter: ImageViewerAdapter

    private var mSize = 0

    private var isShowIndicator = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestedOrientation = ImageViewerSpec.orientation
        setTheme(ImageViewerSpec.theme)
        setContentView(getLayoutId())
        init()
    }

    /**
     * 获取布局
     *
     * @return 返回布局ID
     */
    @LayoutRes
    protected open fun getLayoutId(): Int {
        return R.layout.image_viewer_activity
    }

    /**
     * 初始化
     */
    protected open fun init() {
        tvIndicator = findViewById(R.id.tvIndicator)
        val viewPager = findViewById<ViewPager2>(R.id.viewPager)

        viewPager.registerOnPageChangeCallback(
            object : OnPageChangeCallback() {
                override fun onPageSelected(position: Int) {
                    super.onPageSelected(position)
                    if (isShowIndicator) {
                        updateIndicator(position, mSize)
                    }
                }
            },
        )

        ViewCompat.setTransitionName(viewPager, ImageViewer.SHARED_ELEMENT)

        mAdapter = ImageViewerAdapter(ImageViewerSpec.listData)
        viewPager.adapter = mAdapter

        mAdapter.setOnItemClickListener { _, _ ->
            finishAfterTransition()
        }

        val position = ImageViewerSpec.position
        mSize = mAdapter.itemCount
        if (position >= 0 && mSize > 0) {
            viewPager.setCurrentItem(position, false)
            updateIndicator(position, mSize)
        }

        isShowIndicator = ImageViewerSpec.showIndicator && mSize > 0
        if (isShowIndicator) {
            tvIndicator?.visibility = View.VISIBLE
        }
    }

    /**
     * 更新指示器
     *
     * @param position
     * @param size
     */
    private fun updateIndicator(position: Int, size: Int) {
        tvIndicator?.text = String.format("%s/%s", position.plus(1).coerceAtMost(size), size)
    }

}
